app/Connection.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace App;
+
+use Illuminate\Database\Eloquent\Model;
+use App\User;
+
+class Connection extends Model
+{
+    //Return a list of connections available to you
+    public static function getIncomingConnections(User $user)
+    {
+        $ids = self::where('user_id', $user->id)->pluck('owner_id')->toArray();
+        return array_map(function($id){ return User::find($id); }, $ids);
```

Comment:
You should use `Relationships` and the `Collection` library.

Add a method on `Connection`

```
public function user() 
{
    return $this->belongsTo(User::class, 'user_id');
}

public function owner()
{
    return $this->belongsTo(User::class, 'owner_id');
}
```

Then if you use `$connection->user`, (note: not calling a method, accessing a property) laravel will do some magic and return you an instance of the `User` Class that matches the row with the given ID.

By default calling `->get()` on an Eloquent Query Builder will return you a `Collection` object that has higher order functions. So, you could refactor this whole piece to be

```
return self::where('user_id', $user->id)->get()->map(function ($connection) { return $connection->owner; });
```

Reply:
Cool.